.github/workflows/ci.yml

```diff
@@ -17,28 +56,73 @@ jobs:
         with:
           basic-auth: ${{ secrets.GRAFANA_CLOUD_BASIC_AUTH }}
           hostname: ${{ secrets.GRAFANA_CLOUD_HOST }}
-          this-job-name: Unit Tests ${{ matrix.shard }}
+          this-job-name: Unit Tests ${{ matrix.shardIndex }}
         continue-on-error: true
-
+        
       - name: Checkout the repo
         uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
 
       - name: Setup node
         uses: actions/setup-node@v3
         with:
           node-version: "18"
           cache: yarn
-
+      
       - name: Run yarn install
         run: yarn install --frozen-lockfile
-
+      
       - name: Setup project
         run: yarn setup
         env:
           GH_TOKEN: ${{ github.token }}
-          
+
       - name: Run tests
-        run: yarn run test:ci --shard=${{ matrix.shard }}
-        
-      # - name: Upload coverage
-      #   run: ./bin/codecov -s operator_ui/coverage
+        run: yarn run test:ci --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }}
+
+      - name: Rename unit-test coverage report as belonging to the shard ${{ matrix.shardIndex }}
+        run: mv coverage/lcov.info coverage/${{matrix.shardIndex}}_lcov.info
+      
+      - name: Upload unit-test coverage report
+        if: always()
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+        with:
+          name: unit-tests-coverage
+          path: ./coverage/${{matrix.shardIndex}}_lcov.info
+              
+      - name: Upload CodCov coverage report
+        run: ./bin/codecov -s operator_ui/coverage
+  
+  sonarqube:
+    name: SonarQube Scan
+    needs: [eslint, unit-tests]
+    runs-on: ubuntu-latest
+    if: ${{ always() }}
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        with:
+          fetch-depth: 0 # fetches all history for all tags and branches to provide more metadata for sonar reports
+      
+      - name: Download all reports
+        uses: actions/download-artifact@9782bd6a9848b53b110e712e20e42d89988822b7 # v3.0.1
+
+      - name: Update ESLint paths (for checked files) to source code paths (/home/runner/work/operator-ui/operator-ui/ -> /github/workspace/)
```

Comment:
Nit: I might make this name more succint

Reply:
Shortened, thanks